pennylane/qnn.py

```diff
@@ -0,0 +1,205 @@
+# Copyright 2018-2020 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""This module contains the :class:`~.KerasLayer` class for integrating QNodes with the Keras
+layer API."""
+import functools
+import inspect
+from collections.abc import Iterable
+from typing import Optional
+
+try:
+    import tensorflow as tf
+    from tensorflow.keras.layers import Layer
+    from pennylane.interfaces.tf import to_tf
+
+    CORRECT_TF_VERSION = int(tf.__version__.split(".")[0]) > 1
+except ImportError:
+    from abc import ABC
+
+    Layer = ABC
+    CORRECT_TF_VERSION = False
+
+from pennylane.qnodes import QNode
```

Comment:
I might have missed it, but is this only being imported for the type hinting?

Reply:
Usually it's fine, but I am hesitant doing this for the QNode in PennyLane, as we have had a lot of issues with cyclic imports (and still do). Would prefer not to import it for now if it is just being used for type hints

Reply:
Great, have removed.